File: GooglePay/src/test/java/com/braintreepayments/api/GooglePayExceptionUnitTest.java
package com.braintreepayments.api;

import android.os.Parcel;

import com.google.android.gms.common.api.Status;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static junit.framework.Assert.assertEquals;

@RunWith(RobolectricTestRunner.class)
public class GooglePayExceptionUnitTest {

    @Test
    public void testGooglePayException_isSerializable() {
        Status status = new Status(1, "Some status message");
        GooglePayException exception = new GooglePayException("Some message", status);

        Parcel parcel = Parcel.obtain();
        exception.writeToParcel(parcel, 0);
        parcel.setDataPosition(0);

        GooglePayException actual = GooglePayException.CREATOR.createFromParcel(parcel);

        assertEquals("Some message", actual.getMessage());
        assertEquals("Some status message", actual.getStatus().getStatusMessage());
        assertEquals(1, actual.getStatus().getStatusCode());
    }
}

File: GooglePay/src/test/java/com/braintreepayments/api/googlepay/GooglePayExceptionUnitTest.kt
package com.braintreepayments.api.googlepay

import android.os.Parcel
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import com.google.android.gms.common.api.Status
import kotlinx.parcelize.parcelableCreator
import kotlin.test.assertEquals

@RunWith(RobolectricTestRunner::class)
class GooglePayExceptionUnitTest {

        @Test
        fun `parcels GooglePayException successfully`() {
            val status = Status(1, "Some status message")
            val exception = GooglePayException("Some message", status)

            val parcel = Parcel.obtain().apply {
                exception.writeToParcel(this, 0)
                setDataPosition(0)
            }

            val actual = parcelableCreator<GooglePayException>().createFromParcel(parcel)

            assertEquals("Some message", actual.message)
            assertEquals("Some status message", actual.status?.statusMessage)
            assertEquals(1, actual.status?.statusCode)
        }
}
